test/assets.testsuite.ts

```diff
@@ -79,4 +81,31 @@ export const assetsTests = () => {
     expect(owner).toBe(rootOwner.getAddress());
     expect(offeredTo).toBe(emptyAddress);
   });
+
+  test("update did document for asset", async () => {
+    const assetAddress = await iam.registerAsset();
+
+    const asset1 = await iam.getDidDocument({ did: `did:ethr:${assetAddress}` });
+    expect(asset1.publicKey.length).toBe(1);
+
+    const update = await iam.updateDidDocument({
+      didAttribute: DIDAttribute.PublicKey,
+      did: `did:ethr:${assetAddress}`,
+      data: {
+        algo: Algorithms.Secp256k1,
+        encoding: Encoding.HEX,
+        type: PubKeyType.SignatureAuthentication2018,
+        value: { tag: 'key-1', publicKey: `0x${new Keys().publicKey}` }
+      }
+    });
+    expect(update).toBeTruthy();
+
+    const asset = await iam.getDidDocument({ did: `did:ethr:${assetAddress}` });
+    expect(asset.publicKey.length).toBe(2);
+
+    expect(asset.publicKey[0]).toMatchObject({
```

Comment:
Order of public keys is not defined by spec. Therefore it is more reliable to check that key contained in list

Reply:
Ok, didn't know that. This test passed multiple times, that's why i left it like that :) I will change that.